File: code/Cracking/HardProblem/l3/KnightCircuit--Hard/solve/java/KnightCircuitRunner.java
import java.io.FileReader;
import java.io.FileWriter;

import java.util.List;
import java.util.ArrayList;

import org.topcoder.TopcoderReader;
import org.topcoder.TopcoderWriter;
import org.topcoder.TypeRef;

public class KnightCircuitRunner {
    public static void main(String[] args) {
    try {
        TopcoderReader reader = new TopcoderReader(new FileReader(args[0]));
        int w = (Integer) reader.next(Integer.class);
        reader.next();
        
        int h = (Integer) reader.next(Integer.class);
        reader.next();
        
        int a = (Integer) reader.next(Integer.class);
        reader.next();
        
        int b = (Integer) reader.next(Integer.class);
        reader.close();

        KnightCircuit solver = new KnightCircuit();
        TopcoderWriter writer = new TopcoderWriter(new FileWriter(args[1]));
        writer.write(solver.maxSize(w, h, a, b));
        writer.close();
    } catch (Exception err) {
        err.printStackTrace(System.err);
    }
    }    
}

File: code/Cracking/HardProblem/l3/KnightCircuit--Hard/solve/cpp/KnightCircuitRunner.cpp
#include <topcoder>
using namespace TopCoder;
#include "KnightCircuit.cpp"
#include <fstream>
using namespace std;

int main(int argc, char const *argv[]) {
    try {
        ifstream ifs(argv[1]);
        int w; read(ifs, w); next(ifs);
        int h; read(ifs, h); next(ifs);
        int a; read(ifs, a); next(ifs);
        int b; read(ifs, b);
        ifs.close();

        ofstream ofs(argv[2]);
        show(ofs, maxSize(w, h, a, b));
        ofs.close();
    } catch (exception &e) {
        cerr << e.what() << endl;
    }
    return 0;
}
